### PR TITLE
Deployments pictures: lifecycle and selection algorithm diagram.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: 02.Overview/04.Deployment/docs.md [hunk 1]
```diff
@@ -88,7 +93,8 @@ The values for the status field for a finished update are:
 
 It is possible that you have chosen the device to be the target for one deployment or more.
 When the device is asking for the deployment, the deployments service looks for
-the oldest, not finished deployment for the device.
+the oldest, not finished deployment for the device (see the picture below).
 If there is one, the deployments service creates the instance of the deployment for the device (device deployment) with a "pending" state and executes the following operations:
 * checks if the deployment has more than one phase and in case it does, checks if there is an active phase; if so, the deployments service proceeds, if not, the deployment service returns no instructions to the device;
 * tries to assign artifact to the device; if there is an artifact returns the deployment instructions to the device; if not - returns no instructions and sets the device deployment status to "no artifact"; in case the artifact installed on the device is the same as the one in the deployment, deployments service returns no instructions and sets the device deployment status to "already installed".
+![Select Deployment Algorithm](selectDeploymentForDeviceAlgorithm.png)
```
Comment (mirzak) [inline 02.Overview/04.Deployment/docs.md:100]: To be honest, I have a very hard time following this (both text and diagram). In hindsight I do not think this belongs in "Overview".

But for now I think the addition of the diagram does not make it worse so we can proceed and think about where this should be instead.

**Comment by merlin-northern** [comment]
ok thanks @mirzak ! I corrected inprogress, good catch!
lets do the rest ina follow up PR, as you proposed.

**Comment by mirzak** [comment]
Probably good to rebase this one on top of master (which might resolve the CI errors)

**Comment by merlin-northern** [comment]
ok thanks!
and thanks @drewmoseley for approval as well!

**Pull request merged**